src/mono/mono/mini/interp/transform.c

```diff
@@ -7539,8 +7539,11 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 				g_error ("transform.c: Unimplemented opcode: 0xFE %02x (%s) at 0x%x\n", *td->ip, mono_opcode_name (256 + *td->ip), td->ip-header->code);
 			}
 			break;
-		default:
-			g_error ("transform.c: Unimplemented opcode: %02x at 0x%x\n", *td->ip, td->ip-header->code);
+		default: {
+			g_warning ("opcode 0x%02x not handled", *td->ip);
+			mono_error_set_generic_error (error, "System", "InvalidProgramException", "");
```

Comment:
Can we include the opcode in the message in the exception?
```suggestion
			mono_error_set_generic_error (error, "System", "InvalidProgramException", "opcode 0x%02x not handled", *td->ip);
```